internal/eval/eval.go

```diff
@@ -3522,6 +3528,142 @@ func evalJSONNUMINCRBY(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(resultString, false)
 }
 
+
+func evalDUMP(args []string, store *dstore.Store) []byte {
+    if len(args) < 1 {
+        return diceerrors.NewErrArity("DUMP")
+    }
+    key := args[0]
+    obj := store.Get(key)
+    if obj == nil {
+        return diceerrors.NewErrWithFormattedMessage("-ERR could not perform this operation on a key that doesn't exist")
+    }
+	
+    serializedValue, err := rdbSerialize(obj)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("serialization failed")
+    }
+	encodedResult := base64.StdEncoding.EncodeToString(serializedValue)
+    return clientio.Encode(encodedResult, false)
+}
+
+func evalRestore(args []string, store *dstore.Store) []byte {
+	if len(args) < 2 {
```

Comment:
since the first byte of rdbSerialize is a fixed value `	buf.WriteByte(0x09)` and second is the type, we need atleast 3 as third will be data. 

```suggestion
	if len(args) < 3 {
```